### PR TITLE
Fix: as generator + new build

### PR DESCRIPTION
**Pull request by 0xSwapFeeder**

### CODE REVIEW

**Review by Elli610 — APPROVED** [review]
lgtm. 

CI need to be fixed btw

**Review by gregLibert — CHANGES_REQUESTED** [review]
Please keep the template lietral approach.

File: packages/massa-proto-cli/src/as-gen.ts [hunk 1]
```diff
@@ -2,126 +2,111 @@ import { writeFileSync } from 'fs';
...
+    "${protoData.funcName}",
+    changetype<StaticArray<u8>>(encode${protoData.funcName}Helper(new ${protoData.funcName}Helper(${args.join(', ')}))),
+    coins
+  );${responseDecoding}
 }
+`;
+
+  // Save the content to a ts file
+  writeFileSync(
+    path.join(outputDirectory, `${protoData.funcName}.ts`),
```
Comment (Elli610) [inline packages/massa-proto-cli/src/as-gen.ts:91]: @gregLibert maybe we should call id `${protoData.funcName}Caller.ts`

File: packages/massa-proto-cli/src/protobuf.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ export async function getProtoFunction(protoPath: string): Promise<ProtoFile> {
   // get the arguments of the Helper
   const argFields: FunctionArguments[] = Object.entries(helper.fields)
     .filter(([, value]) => value)
-    .map(([name, field]) => ({ name, type: field.type }));
+    .map(([name, field]) => ({ name, type: field['type'] }));
```
Comment (gregLibert) [inline packages/massa-proto-cli/src/protobuf.ts]: i'm curious, why ?

**Review by gregLibert — APPROVED** [review]

File: packages/massa-proto-cli/src/protobuf.ts (hunk 2 above)
Comment (Elli610) [reply packages/massa-proto-cli/src/protobuf.ts]: the compiler was angry. we don't know exactly why.
Then we updated it like this:

`.map(([name, field]) => ({
      name,
      type: (field as { type: string; id: number }).type,
    }));`

**Comment by gregLibert** [comment]
@Elli610 can you fix the conflicts please ?

**Comment by 0xSwapFeeder** [comment]
> @Elli610 can you fix the conflicts please ?

Branch rebased, deps fixed and new build done!

**Pull request merged**